README.md

```diff
@@ -1,95 +1,108 @@
 
-## Technical Details
+Also, remember that ChoiceScript is by nature not performant, which is fine, given its primary purpose. This library aims to extend ChoiceScript's core functionality, but don't go too crazy with it. Okay? :sweat_smile:
 
-Below follow some extra technical details that will be beneficial to anyone wishing to learn more about the cslib internals, or plan to contribute to its source code.
+Credits are not mandatory but very much appreciated. :pray:
 
-### Global Configuration Variables
 
-You must define all of the following variables in your startup scene (via \*create).
+## Examples:
+### Lowercase
 
-Note that cslib modules (scenes) should not refer to any global variable other than those listed below (or those accessed via a reference passed as a parameter).
+It is very common that mobile keyboards automatically enable uppercase for the first letter of a text input. So you asked your reader to insert a colour for the character's eyes and now, every time, it comes up like this? 
 
-- **cslib_ret** — *anything* (default: "")
-This variable is how all cslib functions return their results. Each function should set this global before \*return with their 'result' (if there is one). If you wish to return more than one value, consider taking the refs (names) of global variables as parameters, which you can assign to via indirect reference.
+> He looks into your sweet *Brown* eyes…
 
-- **cslib_script** — *boolean* (recommended default: false)
-This variable is used to guard against any usage of \*script in cslib modules. If set to true any cslib function *may* switch to using \*script implementations in order to improve performance. Not all functions have \*script implementations, in which case they will continue to use the ChoiceScript variation.
+*Oof!* :joy: What if you could be sure every letter is in lowercase?
 
-- **cslib_assert** — *boolean* (recommended default: false)
-Used to guard cslib asserts.
-If set to true cslib functions are permitted to use \*bug more liberally. Best set to true when testing or debugging, particularly if you're having issues with any cslib code.
+```choicescript
+*gosub_scene cslib_string lowercase eye_colour
+*set eye_colour cslib_ret
+```
 
-### Coding Conventions
+### Highest Stat
 
-Any contributions to the cslib source code should adhere to the following guidelines wherever possible.
+Very often authors need to check which of the stats is the highest one, whether to display flavour text or decide which branch to follow in the story. Writing a loop in ChoiceScript is not trivial and some authors even prefer to write many lines of `*if` statements just to work around it. But what if you could check it in a single line of code?!
 
-#### Coding
+```
+*gosub_scene cslib_number max_stat "strength" "dexterity" "constitution" "intelligence" "wisdom" "charisma"
```

Comment:
ADD: Please note the use of quotes around the variable names, which are dereferenced in the function. The return value is also a string, which you can dereference to get the actual value of the property.

Reply:
Here's only an example, the user should still check the documentation. But I agree we can add these notes.

Reply:
Agreed, let's keep the README light and fluffy! But we should discuss dereferencing in some docs somewhere.